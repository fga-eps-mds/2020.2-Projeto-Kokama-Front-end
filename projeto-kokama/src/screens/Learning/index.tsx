import {
    View,
    Text,
    Image,
    ScrollView,
    SafeAreaView,
    TouchableWithoutFeedback,
} from "react-native";
import React from "react";
import cardsStyle from "./styles";

export default function Learning({ navigation }) {
    return (
        <SafeAreaView>
            <ScrollView>
                <View style={{ alignItems: "flex-start" }}>
                    <Text style={cardsStyle.cardTitle}>Histórias</Text>
                    <TouchableWithoutFeedback testID = 'history'
                        onPress={() => navigation.push('Histórias do Povo Kokama')}>
                        <Image style={cardsStyle.historyCard} source={require("../../assets/img/floresta.jpg")} />
                    </TouchableWithoutFeedback>
                    <Text style={cardsStyle.cardTitle}>Atividades</Text>
                    <TouchableWithoutFeedback testID = 'activity'
                        onPress={() => navigation.push('Atividades')}>
                        <Image style={cardsStyle.historyCard} source={require("../../assets/img/study.jpg")} />
                    </TouchableWithoutFeedback>
                </View>
            </ScrollView>
        </SafeAreaView>
    );
}