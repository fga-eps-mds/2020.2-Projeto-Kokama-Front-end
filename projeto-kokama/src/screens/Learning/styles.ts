import { StyleSheet, Dimensions, Platform, StatusBar } from "react-native";
import Colors from "../../assets/Colors";

// Constants and Global Variables
const window = Dimensions.get("window");
const screen = Dimensions.get("screen");

const styles = StyleSheet.create({
    container: {
        maxWidth: screen.width,
        height: window.height,
        backgroundColor: Colors.LIGHT_GRAY,
    },
    historyText: {
        textAlign: "left",
        fontSize: 20,
        paddingLeft: 23,
        color: Colors.TEXT,
    },
    historyCard: {
        height: 150,
        width: screen.width,
    },
});

export default styles;