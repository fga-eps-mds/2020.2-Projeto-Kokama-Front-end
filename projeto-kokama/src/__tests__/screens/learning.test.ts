import renderer, { act } from 'react-test-renderer';
import Learning from '../../screens/Learning';

test("Check Learning screen", () => {

    let learning = Learning({navigation: { navigator }})
    let tree = renderer.create(learning)
    expect(tree).toMatchSnapshot();
    // let TouchableWithoutFeedback = tree.root.findByProps({testID:'history'}).props;
    // act(() => TouchableWithoutFeedback.onPress());
})

