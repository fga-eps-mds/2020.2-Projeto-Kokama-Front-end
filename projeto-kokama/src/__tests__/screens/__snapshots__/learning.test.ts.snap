// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Learning screen 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#333333",
              "fontSize": 20,
              "marginTop": 10,
              "paddingLeft": 23,
              "textAlign": "left",
            }
          }
        >
          Hist√≥rias
        </Text>
        <Image
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          source={1}
          style={
            Object {
              "height": 150,
              "width": 750,
            }
          }
          testID="history"
        />
        <Text
          style={
            Object {
              "color": "#333333",
              "fontSize": 20,
              "marginTop": 10,
              "paddingLeft": 23,
              "textAlign": "left",
            }
          }
        >
          Atividades
        </Text>
        <Image
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          source={1}
          style={
            Object {
              "height": 150,
              "width": 750,
            }
          }
          testID="activity"
        />
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
